src/commands/server/update.ts

```diff
@@ -182,65 +173,148 @@ Consider removing '--che-operator-image' to update Eclipse Che operator to the s
+        merge(resetImagesCrPatch, { spec: { server: { cheImageTag: '' } } })
       }
-      ctx[ChectlContext.CR_PATCH] = crPatch
 
       if (imagesListMsg) {
-        cli.warn(`In order to update Eclipse Che to a newer version the fields defining the images in the '${cheCluster.metadata.name}'
-Custom Resource in the '${flags.chenamespace}' namespace will be cleaned up:${imagesListMsg}`)
-        if (!flags.yes && !await cli.confirm('If you want to continue - press Y')) {
-          cli.info('Update cancelled by user.')
-          this.exit(0)
+        cli.warn(`Custom images found in '${cheCluster.metadata.name}' Custom Resource in the '${flags.chenamespace}' namespace: ${imagesListMsg}`)
+        if (flags.batch || flags.yes || await cli.confirm('Do you want to preserve custom images [y/n]?')) {
+          cli.info('Keeping current images.\nNote, Update might fail if functionality of the custom images different from the default ones.')
+        } else {
+          cli.info('Resetting custom images to default ones.')
+
+          const ctx = ChectlContext.get()
+          const crPatch = ctx[ChectlContext.CR_PATCH] || {}
+          merge(crPatch, resetImagesCrPatch)
+          ctx[ChectlContext.CR_PATCH] = crPatch
         }
       }
     }
+  }
 
-    try {
-      await updateTasks.run(ctx)
-      await postUpdateTasks.run(ctx)
+  /**
+   * Check whether chectl should proceed with update.
+   * Asks user for confirmation (unless assume yes is provided).
+   * Is applicable to operator installer only.
+   * Returns true if chectl can/should proceed with update, false otherwise.
+   */
+  private async checkAbilityToUpdateCheOperatorAndAskUser(flags: any): Promise<boolean> {
+    const ctx = ChectlContext.get()
+    cli.info(`Existing Eclipse Che operator: ${ctx.deployedCheOperatorImage}`)
+    cli.info(`New Eclipse Che operator     : ${ctx.newCheOperatorImage}`)
 
-      this.log(getCommandSuccessMessage())
-    } catch (err) {
-      this.error(getCommandErrorMessage(err))
+    if (ctx.deployedCheOperatorImageName === DEFAULT_CHE_OPERATOR_IMAGE_NAME && ctx.newCheOperatorImageName === DEFAULT_CHE_OPERATOR_IMAGE_NAME) {
+      // Official images
+
+      if (ctx.deployedCheOperatorImage === ctx.newCheOperatorImage) {
+        if (ctx.newCheOperatorImageTag === 'nightly') {
+          cli.info('Updating current Eclipse Che nightly version to a new one.')
+          return true
+        }
+
+        if (flags[CHE_OPERATOR_CR_PATCH_YAML_KEY]) {
+          // Despite the operator image is the same, CR patch might contain some changes.
+          cli.info('Patching existing Eclipse Che installation.')
+          return true
+        } else {
+          cli.info('Eclipse Che is already up to date.')
+          return false
+        }
+      }
+
+      if (ctx.newCheOperatorImageTag === 'nightly' || (ctx.deployedCheOperatorImageTag !== 'nightly' && semver.gt(ctx.newCheOperatorImageTag, ctx.deployedCheOperatorImageTag))) {
```

Comment:
This condition assumes that the user does an update if new tag is `nightly` (we handle the case with both `nightly` tags above) OR if old tag IS NOT nightly (because it would be downgrade) AND new tag is greater than old one.